### PR TITLE
add silent mode support for vector

### PR DESCRIPTION
**Pull request by cperkulator**
haven't tested on hardware yet

### CODE REVIEW

File: can/interfaces/vector/canlib.py [hunk 1]
```diff
@@ -204,6 +204,8 @@ def __init__(
         self.channel_masks: Dict[int, int] = {}
         self.index_to_channel: Dict[int, int] = {}
         self._can_protocol = CanProtocol.CAN_FD if is_fd else CanProtocol.CAN_20
+        
+        self.silent_mode = kwargs.get('silent_mode', False)
```
Comment (zariiii9003) [inline can/interfaces/vector/canlib.py:208]: I'd prefer `listen_only` like in socketcan. Also, make this variable private.

**Comment by zariiii9003** [comment]
Are you still working on the failing tests?

**Comment by cperkulator** [comment]
@zariiii9003 looks like tests are passing except one that is known to periodically fail

**Review by zariiii9003 — APPROVED** [review]
Looks good, thank you

**Comment by cperkulator** [comment]
@zariiii9003 is this ready to get merged in?

**Pull request merged**